### ISSUE TITLE
build failure copperspice-1.4.2

### ISSUE THREAD

**Issue by ctl83** [copperspice/copperspice#57]
Hi there,
I can't build copperspice-1.4.2 on slackware64-14.2 with cups-2.1.4
Here is the error I get:

```
src/gui/dialogs/qprintdialog_unix.cpp: In member function 'void QUnixPrintWidgetPrivate::setCupsProperties()':
src/gui/dialogs/qprintdialog_unix.cpp:1000:51: error: invalid user-defined conversion from 'QByteArray' to 'const char*' [-fpermissive]
       QRect pageRect = cups->pageRect(cupsPageSize);
                                                   ^
In file included from include/QtCore/qvariant.h:30:0,
                 from include/QtCore/csmeta_internal_1.h:27,
                 from include/QtCore/csmeta.h:26,
                 from include/QtCore/qmetaobject.h:29,
                 from include/QtCore/qobject.h:31,
                 from include/QtGui/qwidget.h:28,
                 from privateinclude/QtGui/private/qwidget_p.h:26,
                 from privateinclude/QtGui/private/qdialog_p.h:26,
                 from privateinclude/QtGui/private/qabstractprintdialog_p.h:26,
                 from src/gui/dialogs/qprintdialog_unix.cpp:27:
include/QtCore/qbytearray.h:383:4: note: candidate is: QByteArray::operator QNoImplicitBoolCast() const <near match>
    operator QNoImplicitBoolCast() const;
    ^
include/QtCore/qbytearray.h:383:4: note:   no known conversion from 'QNoImplicitBoolCast {aka int}' to 'const char*'
include/QtCore/qbytearray.h:383:4: error: 'QByteArray::operator QNoImplicitBoolCast() const' is private
src/gui/dialogs/qprintdialog_unix.cpp:1000:51: error: within this context
       QRect pageRect = cups->pageRect(cupsPageSize);
                                                   ^
In file included from src/gui/dialogs/qprintdialog_unix.cpp:44:0:
privateinclude/QtGui/private/qcups_p.h:65:10: note:   initializing argument 1 of 'QRect QCUPSSupport::pageRect(const char*) const'
    QRect pageRect(const char *choice) const;
          ^
src/gui/dialogs/qprintdialog_unix.cpp:1003:53: error: invalid user-defined conversion from 'QByteArray' to 'const char*' [-fpermissive]
       QRect paperRect = cups->paperRect(cupsPageSize);
                                                     ^
In file included from include/QtCore/qvariant.h:30:0,
                 from include/QtCore/csmeta_internal_1.h:27,
                 from include/QtCore/csmeta.h:26,
                 from include/QtCore/qmetaobject.h:29,
                 from include/QtCore/qobject.h:31,
                 from include/QtGui/qwidget.h:28,
                 from privateinclude/QtGui/private/qwidget_p.h:26,
                 from privateinclude/QtGui/private/qdialog_p.h:26,
                 from privateinclude/QtGui/private/qabstractprintdialog_p.h:26,
                 from src/gui/dialogs/qprintdialog_unix.cpp:27:
include/QtCore/qbytearray.h:383:4: note: candidate is: QByteArray::operator QNoImplicitBoolCast() const <near match>
    operator QNoImplicitBoolCast() const;
    ^
include/QtCore/qbytearray.h:383:4: note:   no known conversion from 'QNoImplicitBoolCast {aka int}' to 'const char*'
include/QtCore/qbytearray.h:383:4: error: 'QByteArray::operator QNoImplicitBoolCast() const' is private
src/gui/dialogs/qprintdialog_unix.cpp:1003:53: error: within this context
       QRect paperRect = cups->paperRect(cupsPageSize);
                                                     ^
In file included from src/gui/dialogs/qprintdialog_unix.cpp:44:0:
privateinclude/QtGui/private/qcups_p.h:64:10: note:   initializing argument 1 of 'QRect QCUPSSupport::paperRect(const char*) const'
    QRect paperRect(const char *choice) const;
          ^
Makefile:19214 : la recette pour la cible « src/gui/dialogs/lib_libCsGui1.4_la-qprintdialog_unix.lo » a échouée
make[2]: *** [src/gui/dialogs/lib_libCsGui1.4_la-qprintdialog_unix.lo] Erreur 1
```

**Comment by mortaromarcello** [comment]
Change `QRect pageRect = cups->pageRect(cupsPageSize);` with `QRect pageRect = cups->pageRect(cupsPageSize.data());`

**Comment by agserm** [comment]
The fix for this has been merged in 3be413f. Thanks to you both for the report and the pull request!
